### PR TITLE
Added Search Terms for Searchable Pages

### PR DESCRIPTION
**Pull request by aletail**
https://werkbotstudios.teamwork.com/app/tasks/35150248

### Testing Steps
- [ ] Test with United Way: https://github.com/werkbot/united-way-erie/tree/feature/search-terms
- [ ] dev/build
- [ ] Edit a page, under settings you can define search terms
- [ ] Test search on the front end, confirm the search terms help with results

### Issues/Concerns
- I decided to only have these search terms on the pages, so they reside in one spot. It is flexible enough that we can change this in the future, but I think this should work for most use cases

### Git Flow
- **DO NOT** delete "release/\*" or "hotfix/\*" branches after merging a PR. These are used to publish the next release, and they are deleted automatically.
- "Squash and merge" is good on "feature/\*" into "develop"
- "Create a merge commit" is good on "release/\*" or "hotfix/\*" into "main"

### CODE REVIEW

File: src/SearchableExtension.php [hunk 1]
```diff
@@ -19,6 +41,52 @@ class SearchableExtension extends DataExtension
...
+
+  /**
+   * updateSettingsFields
+   * Adds the SearchTerms GridField to the settings tab
+   * This should only be applied to SiteTree objects
+   *
+   * @param FieldList $fields
+   * @return void
+   **/
+  public function updateSettingsFields(FieldList $fields)
```
Comment (tiller1010) [inline src/SearchableExtension.php:53]: This being under settings makes sense for pages, but for other searchable dataobjects, like articles and content layouts, they do not have settings fields.

**Review by tiller1010 — APPROVED** [review]
> I decided to only have these search terms on the pages, so they reside in one spot. It is flexible enough that we can change this in the future, but I think this should work for most use cases

Approving, but I left a note about using settingsFields

File: src/SearchableExtension.php (hunk 1 above)
Comment (aletail) [reply src/SearchableExtension.php:68]: @tiller1010 Take a look at this again, I updated so the search terms are added to all objects

Comment (aletail) [reply src/SearchableExtension.php:68]: I forgot to mention the content layout, blog, etc....need their getIndexQuery functions updated to include search terms. I figure I would do that after this is merged and released

**Review by tiller1010 — APPROVED** [review]
Nice!

**Pull request merged**